cmd/prometheus/main.go

```diff
@@ -1005,3 +1018,52 @@ func (opts tsdbOptions) ToTSDBOptions() tsdb.Options {
 		MaxBlockDuration:       int64(time.Duration(opts.MaxBlockDuration) / time.Millisecond),
 	}
 }
+
+func initTracing(logger log.Logger) (io.Closer, error) {
+	// Set tracing configuration defaults
+	cfg := &jaegercfg.Configuration{
+		ServiceName: "prometheus",
+		Disabled:    true,
+		Sampler: &jaegercfg.SamplerConfig{
+			Type:  jaeger.SamplerTypeConst,
+			Param: 1,
+		},
+		Reporter: &jaegercfg.ReporterConfig{
+			LogSpans: false,
+		},
+	}
+
+	// Override defaults from environment variables. Available options can be seen here:
```

Comment:
I'm not sure about this mix of hardcoded and environment. Are there not sensible defaults?

Reply:
The comment is wrong

Reply:
You need at least to be able to enable it, set your jaeger server, your extra tags, etc

Reply:
- The `ServiceName` is required.
- Tracing is enabled by default, I think that should definitely be flipped around.
- Without a sampler config, nothing is captured, so I think it's a sensible default for a feature targeted at developers.
- `Reporter` can be left out.

With this
```go
jaegercfg.Configuration{
  ServiceName: "prometheus",
  Disabled:    true,
}
```
the env vars to set are:
```bash
JAEGER_DISABLED=false 
JAEGER_SAMPLER_TYPE=const 
JAEGER_SAMPLER_PARAM=1 
JAEGER_ENDPOINT=http://jaeger.local
```

Reply:
That seems reasonably minimal.

Reply:
Updated to use minimum config in code